tensorflow/contrib/seq2seq/python/ops/beam_search_decoder.py

```diff
@@ -297,11 +298,24 @@ def initialize(self, name=None):
     """
     finished, start_inputs = self._finished, self._start_inputs
 
+    log_prob_mask = array_ops.one_hot(  # shape(batch_sz, beam_sz)
+        array_ops.zeros([self._batch_size], dtype=dtypes.int32),
+        depth=self._beam_width, on_value=True,
+        off_value=False, dtype=dtypes.bool)
+
+    log_prob_zeros = array_ops.zeros([self._batch_size, self._beam_width],
+                                     dtype=nest.flatten(
+                                         self._initial_cell_state)[0].dtype)
+    log_prob_neg_inf = array_ops.ones([self._batch_size, self._beam_width],
+                                      dtype=nest.flatten(
+                                          self._initial_cell_state)[0].dtype
+                                     ) * -np.Inf
+
+    log_probs = array_ops.where(log_prob_mask, log_prob_zeros, log_prob_neg_inf)
```

Comment:
I think you don't need the `log_prob_mask`, `log_prob_zeros`, and `log_prob_neg_inf` here.

simply set
```
log_probs = one_hot(zeros([batch_size]), depth=beam_width, on_value=0.0, off_value=-np.Inf)
```
should be enough. please also update the test.